browsergym/experiments/src/browsergym/experiments/loop.py

```diff
@@ -657,6 +658,75 @@ def summary_info(self) -> dict:
                 self._summary_info = json.load(f)
         return self._summary_info
 
+    def get_tape(self) -> Tape:
+        """
+        Return experiment trace in a format compatible with the TapeAgents framework
+        """
+        tape_steps = []
+        for step_info in self.steps_info:
+            # extract observation step
+            if step_info.obs is not None:
+                screenshot: str = ""
+                screenshot_som: str = ""
+                if "screenshot" in step_info.obs:
+                    screenshot = f"{self.exp_dir}/screenshot_step_{step_info.step}.png"
+                    step_info.obs.pop("screenshot")
+                if "screenshot_som" in step_info.obs:
+                    screenshot_som = f"{self.exp_dir}/screenshot_som_step_{step_info.step}.png"
+                    step_info.obs.pop("screenshot_som")
+                observation = Observation(
+                    metadata={"step": step_info.step},
+                    obs=step_info.obs,
+                    screenshot=screenshot,
+                    screenshot_som=screenshot_som,
+                )
+                tape_steps.append(observation)
+
+            # extract thought step
```

Comment:
We should probably look if there is a tape_sgement first, like we discussed. Or should we wait until it's implemented?

Reply:
Could a tape_segment simply be a Tape that can be appened

Reply:
We could merge this implementation to support existing traces and work on the segments separately. I think that segments would be either Tape or list[Step], cause trace metadata is not really needed in a segment

Reply:
I've added support for tape_segment dicts stored in the `other_info` dict of the agent info.